core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -770,6 +770,36 @@ object PSBT extends Factory[PSBT] {
     PSBT(globalMap, inputMaps, outputMaps)
   }
 
+  /**
+    * Creates a PSBT with data from the transaction
+    * @param tx global transaction for the PSBT
+    * @return Created PSBT
+    */
+  def fromTx(tx: Transaction): PSBT = {
+
+    val unsignedInputs = tx.inputs.map(
+      input =>
+        TransactionInput(input.previousOutput,
+                         EmptyScriptSignature,
+                         input.sequence))
+    val unsignedTx =
+      BaseTransaction(tx.version, unsignedInputs, tx.outputs, tx.lockTime)
+
+    val globalMap = GlobalPSBTMap(
+      Vector(GlobalPSBTRecord.UnsignedTransaction(unsignedTx)))
+    val inputMaps = {
+      tx match {
+        case btx: BaseTransaction =>
+          btx.inputs.map(InputPSBTMap.fromTransactionInput).toVector
+        case wtx: WitnessTransaction =>
+          wtx.witness.witnesses.map(InputPSBTMap.fromWitness)
```

Comment:
Does this cover the `P2SH(P2WSH)` case correctly?

Reply:
It should, those qualify as a `WitnessTransaction`